cmd/surf/commands/createextractor.go

```diff
@@ -2,59 +2,77 @@ package commands
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"github.com/CloudHub360/ch360.go/ch360"
 	"github.com/CloudHub360/ch360.go/config"
+	"github.com/pkg/errors"
 	"io"
 	"os"
 )
 
-//go:generate mockery -name "ExtractorCreator"
-
 const CreateExtractorCommand = "create extractor"
 
-type ExtractorCreator interface {
-	Create(ctx context.Context, name string, config io.Reader) error
-	CreateFromModules(ctx context.Context, name string, config io.Reader) error
-}
-
 type CreateExtractor struct {
 	writer        io.Writer
 	creator       ExtractorCreator
 	extractorName string
-	config        io.Reader
+	template      *ch360.ModulesTemplate
 }
 
 func NewCreateExtractor(writer io.Writer,
 	creator ExtractorCreator,
 	extractorName string,
-	config io.Reader) *CreateExtractor {
+	template *ch360.ModulesTemplate) *CreateExtractor {
 	return &CreateExtractor{
 		writer:        writer,
 		creator:       creator,
-		config:        config,
+		template:      template,
 		extractorName: extractorName,
 	}
 }
 
-func NewCreateExtractorFromArgs(params *config.RunParams,
+func NewCreateExtractorWithArgs(params *config.RunParams,
 	client ExtractorCreator, out io.Writer) (*CreateExtractor, error) {
 
-	configFile, err := os.Open(params.ConfigPath)
-	if err != nil {
-		return nil, errors.New(fmt.Sprintf("The file '%s' could not be found.", params.ConfigPath))
+	var (
+		err          error
+		templateFile *os.File
+		template     = new(ch360.ModulesTemplate)
+	)
+
+	if params.ModulesTemplate != "" {
+		// deserialise template file
+		templateFile, err = os.Open(params.ModulesTemplate)
+
+		if err != nil {
+			return nil, err
+		}
+
+		template, err = ch360.NewModulesTemplateFromJson(templateFile)
+
+		if err != nil {
+			return nil, errors.WithMessagef(err, "Error when reading json template '%s'", params.ModulesTemplate)
+		}
+
+	} else {
+		// create template struct from IDs
```

Comment:
✋ I don't see a checks (I assume should be round about here) for whether the user has not specified any modules, or the same module more than once. I think those are easy enough that we should check for them before submitting the template rather than relying on the service. It will be easier to give a nice user-friendly error aswell.

Reply:
As discussed on slack - we decided not to add the check for duplicate modules, since we'd break surf if we then allowed them in the api. 

Docopt should also prevent the case where no module ids are specified on the cmdline.